partiql-eval/src/eval/evaluable.rs

```diff
@@ -1009,30 +1009,26 @@ impl EvalOrderBy {
                 let r = spec.expr.evaluate(&r, ctx);
 
                 match spec.spec {
-                    EvalOrderBySortSpec::AscNullsFirst => l.as_ref().cmp(r.as_ref()),
-                    EvalOrderBySortSpec::AscNullsLast => match (l.as_ref(), r.as_ref()) {
-                        (Null, Null) => Ordering::Equal,
-                        (Null, Missing) => Ordering::Less,
-                        (Missing, Missing) => Ordering::Equal,
-                        (Missing, Null) => Ordering::Greater,
-                        (Null, _) => Ordering::Greater,
-                        (Missing, _) => Ordering::Greater,
-                        (_, Null) => Ordering::Less,
-                        (_, Missing) => Ordering::Less,
-                        (l, r) => l.cmp(r),
-                    },
-                    EvalOrderBySortSpec::DescNullsFirst => match (l.as_ref(), r.as_ref()) {
-                        (Null, Null) => Ordering::Equal,
-                        (Null, Missing) => Ordering::Less,
-                        (Missing, Missing) => Ordering::Equal,
-                        (Missing, Null) => Ordering::Greater,
-                        (Null, _) => Ordering::Less,
-                        (Missing, _) => Ordering::Less,
-                        (_, Null) => Ordering::Greater,
-                        (_, Missing) => Ordering::Greater,
-                        (l, r) => r.cmp(l),
-                    },
-                    EvalOrderBySortSpec::DescNullsLast => r.as_ref().cmp(l.as_ref()),
+                    EvalOrderBySortSpec::AscNullsFirst => {
+                        let wrap = NullSortedValue::<true, Value>;
+                        let (l, r) = (wrap(l.as_ref()), wrap(r.as_ref()));
+                        l.cmp(&r)
+                    }
+                    EvalOrderBySortSpec::AscNullsLast => {
+                        let wrap = NullSortedValue::<false, Value>;
+                        let (l, r) = (wrap(l.as_ref()), wrap(r.as_ref()));
+                        l.cmp(&r)
+                    }
+                    EvalOrderBySortSpec::DescNullsFirst => {
+                        let wrap = NullSortedValue::<false, Value>;
+                        let (l, r) = (wrap(l.as_ref()), wrap(r.as_ref()));
+                        r.cmp(&l)
+                    }
+                    EvalOrderBySortSpec::DescNullsLast => {
+                        let wrap = NullSortedValue::<true, Value>;
+                        let (l, r) = (wrap(l.as_ref()), wrap(r.as_ref()));
+                        r.cmp(&l)
```

Comment:
These are doing the correct ordering with respect to null/missing values.

For `DescNullsFirst`, it'll do the full ordering with nulls assigned to be `Ordering::Greater` (i.e. the false). Since we do `r` compared to `l`, the result is flipped, so this will order in descending order with the null/missing values first.

For `DescNullsLast`, it'll do the full ordering with nulls assigned to be `Ordering::Less` (i.e. the true). And since we do `r` compared to `l`, the result is flipped. This will order in descending order with the null/missing values last.

I noticed these were not part of the conformance tests, so added some tests in this PR: https://github.com/partiql/partiql-tests/pull/106.